11-embeddings-reranker-classification-tensorrt/BEI-baai-bge-en-icl-embedding/README.md

```diff
@@ -0,0 +1,157 @@
+# Baseten-Embeddings-Inference with BAAI/bge-en-icl-embedding
+
+This is a Deployment for BEI (Baseten-Embeddings-Inference) with BAAI/bge-en-icl-embedding. BEI is Baseten's solution for production-grade deployments via TensorRT-LLM.
+
+With BEI you get the following benefits:
+- *lowest-latency inference* across any embedding solution (vLLM, SGlang, Infinity, TEI, Ollama)*1
+- *highest-throughput inference* across any embedding solution (vLLM, SGlang, Infinity, TEI, Ollama) - thanks to XQA kernels, FP8 and dynamic batching.*2
+- high parallelism: up to 1400 client embeddings per second
+- cached model weights for fast vertical scaling and high availability - no Hugging Face hub dependency at runtime
+
+# Examples:
+This deployment is specifically designed for the Hugging Face model [BAAI/bge-en-icl](https://huggingface.co/BAAI/bge-en-icl).
+It will also work for fine-tuned models that have the architecture of MistralModel specified in their Hugging Face transformers config.
+Suitable models need to have the configurations of the `sentence-transformers` library, which are used for embeddings. Such repos contain e.g. a `sbert_config.json` or a `1_Pooling/config.json` file besides the fast-tokenizer and the safetensors file.
+
+BAAI/bge-en-icl  is a text-embeddings model, producing a 1D embeddings vector, given an input.
+It's frequently used for downstream tasks like clustering, used with vector databases.
+
+This model is quantized to FP8 for deployment, which is supported by Nvidia's newest GPUs e.g. H100, H100_40B or L4. Quantization is optional, but leads to higher efficiency.
```

Comment:
```suggestion
This model is quantized to FP8 for deployment, which is supported by Nvidia's newest GPUs e.g. H100, H100_40GB or L4. Quantization is optional, but leads to higher efficiency.
```

Reply:
Does quantization cause quality loss on smaller embeddings models?

Reply:
Quantization causes always some loss - In some cases it's not noticeable. 

For embedding models you measure correctness with cosine similarity. Cosine similarity (scale 0-1) is pretty tolerant to quantization. 

The required cosine similarity for a good vector databases is ~0.95, above generates likely no improvements for HNSW algorithm. We get a worse case ~0.99 cosine similarity for fp16 and ~0.985 for fp8 on H100. 

Smaller embedding models are scarce (people did not train a model on Llama-3-1B yet!!), and no fp8 support for bert.